pkg/kernel/tools/heal/liveness_check.go

```diff
@@ -33,58 +35,140 @@ const (
 	packetCount    = 4
 )
 
-// KernelLivenessCheck is an implementation of heal.LivenessCheck. It sends ICMP
-// ping and checks reply. Returns false if didn't get reply.
-func KernelLivenessCheck(deadlineCtx context.Context, conn *networkservice.Connection) bool {
-	if mechanism := conn.GetMechanism().GetType(); mechanism != kernel.MECHANISM {
-		log.FromContext(deadlineCtx).Warnf("ping is not supported for mechanism %v", mechanism)
-		return true
-	}
+type options struct {
+	pingerFactory PingerFactory
+}
+
+// Option is an option pattern for LivelinessChecker
+type Option func(o *options)
 
-	deadline, ok := deadlineCtx.Deadline()
-	if !ok {
-		deadline = time.Now().Add(defaultTimeout)
+// WithPingerFactory - sets any custom pinger factory
+func WithPingerFactory(pf PingerFactory) Option {
+	return func(o *options) {
+		o.pingerFactory = pf
 	}
+}
 
-	addrCount := len(conn.GetContext().GetIpContext().GetDstIpAddrs())
-	if addrCount == 0 {
-		log.FromContext(deadlineCtx).Debug("No dst IP address")
-		return true
+// KernelLivenessCheck returns an implementation of heal.LivenessCheck. It sends ICMP
+// ping and checks reply. Returns false if didn't get reply.
+func KernelLivenessCheck(opts ...Option) heal.LivenessCheck {
```

Comment:
Seems to me this change will break backward compatibility.

We might need TODO:

1. ask community folks.
2. do not change signature of KernelLivenessCheck function.

Reply:
Chose option 2. Fixed